springwolf-plugins/springwolf-cloud-stream-plugin/src/main/java/io/github/springwolf/plugins/cloudstream/asyncapi/scanners/channels/CloudStreamFunctionChannelsScanner.java

```diff
@@ -96,8 +100,18 @@ private Map<String, MessageBinding> buildMessageBinding() {
         return Map.of(protocolName, new EmptyMessageBinding());
     }
 
-    private Map<String, ChannelBinding> buildChannelBinding() {
+    private Map<String, ChannelBinding> buildChannelBinding(Annotation annotation) {
```

Comment:
This is the right place for the change :)

For the other scanners, we have written some utility to extract the i.e. `@KafkaAsyncOperationBinding` using an OperationBindingProcessor.

In this method, a call to the utility (adapted to channelBinding) is sufficient
```java
Map<String, OperationBinding> operationBinding =
                AsyncAnnotationUtil.processOperationBindingFromAnnotation(method, operationBindingProcessors);
Map<String, OperationBinding> opBinding = operationBinding != null ? new HashMap<>(operationBinding) : null;
```
Source: AsyncAnnotationScanner